tests/acceptance/ReinstallFromScratchCest.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace MailPoet\Test\Acceptance;
+
+use MailPoet\Models\Setting;
+use MailPoet\Test\DataFactories\Form;
+use MailPoet\Test\DataFactories\Newsletter;
+use MailPoet\Test\DataFactories\Segment;
+use MailPoet\Test\DataFactories\Subscriber;
+use PHPUnit_Framework_Assert as Asserts;
+
+require_once __DIR__ . '/../DataFactories/Form.php';
+require_once __DIR__ . '/../DataFactories/Newsletter.php';
+require_once __DIR__ . '/../DataFactories/Segment.php';
+require_once __DIR__ . '/../DataFactories/Subscriber.php';
+
+// This test should run last, because it removes all test data
+class ReinstallFromScratchCest {
+
+  function reinstallFromScratch(\AcceptanceTester $I) {
+    $I->wantTo('Reinstall from scratch');
+    $I->login();
+
+    // Step 1 - create email, form, list and subscribers
+    $newsletter = new Newsletter();
+    $newsletter->create();
+    $form = new Form();
+    $form->create();
+    $segment = new Segment();
+    $segment->create();
+    $subscriber = new Subscriber();
+    $subscriber->create();
+    // Create few WP users, which should be imported after reinstall
+    for($i = 0; $i <= 5; $i++) {
+      wp_create_user('test' . $i, 'password', 'imported' . $i . '@from.wordpress');
+    }
+
+    // Step 2 - reinstall from scratch
+    $I->amOnPage('/wp-admin/admin.php?page=mailpoet-settings#advanced');
+    $I->waitForElement('#mailpoet_reinstall', 5);
```

Comment:
I don't think 5 seconds is enough. We tend to use numbers over 20 here. And I think there is no downside to use bigger numbers. It removes the test flakiness.